### DISCUSSION TITLE
Page interaction on click for pagination

### DISCUSSION THREAD

**Discussion by tonge495** [twilio-labs/paste#3222] [Q&A] [answered]
I am working on some pages where there are article search results that require pagination.  Is there any specific guidance around whether we automatically put the user to the top of the page when they click on one of the buttons or if we scroll the user to the top of the page?  See below for examples of both:

- [Option 1: automatic snap to top](https://www.awesomescreenshot.com/video/17239014?key=a1356ee2e4d333fa255a830fbc85d2e2)
- [Option 2: autoscroll](https://www.awesomescreenshot.com/video/17239052?key=c1ef2ddebc5371b2ba9cb48d10334760)

Any advice would be super helpful, thank you!

**Comment by SiTaggart** [comment, marked as answer]
Hi @tonge495 

I don't think we have a particular preference, it's largely based on user expectations based on what you're actually doing, aligning to default browser behaviours.

There are largely two approaches:

1) Each page in the pagination takes the user to a new route/url. When visiting that route the table of results will always be the same. That way you can share your search query and results (Preferred). In this case the action is a page navigation, just like navigating to any other page, and should act accordingly. This usually takes a user to the very top of the page.

2) Each page in the pagination just re-renders the table above. No route change. In this case you must manually place the user above the table of results to inform them that something on the page has happened. 

In both cases, whether you jump or soft scroll to the new location is up to you, and potentially what action is being taken in the browser, will dictate what you _can_ do.

Hope that helps a little.
